File: vhdl_toolkit/vhdlConvertor/hdlObjects/Port.java
package hdlObjects;

import org.json.JSONException;
import org.json.JSONObject;

public class Port implements iJsonable {
	public Direction direction;
	public Variable variable;

	public JSONObject toJson() throws JSONException {
		JSONObject v = new JSONObject();
		v.put("direction", direction);
		v.put("variable", variable.toJson());
		return v;
	}
}

File: hdl_toolkit/parser/hdlConvertor/src/hdlObjects/port.cpp
#include "port.h"

PyObject * Port::toJson() const {
	PyObject *d = PyDict_New();
	PyDict_SetItemString(d, "direction",
			PyUnicode_FromString(Direction_toString(direction)));
	PyDict_SetItemString(d, "variable", variable->toJson());
	return d;
}
